### PR TITLE
Make google_container_cluster work for both routes-based and autopilot clusters

### PR DESCRIPTION
**Pull request by ScottSuarez**
takes over change https://github.com/GoogleCloudPlatform/magic-modules/pull/5540

closes https://github.com/hashicorp/terraform-provider-google/issues/10782
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
@rileykarson is this the solution you had in mind? 
https://github.com/GoogleCloudPlatform/magic-modules/pull/5540#pullrequestreview-830572923

**Comment by davidwallacejackson** [comment]
Just pinging here since I encountered this bug today and will have to use an odd workaround or downgrade the provider -- can somebody review this?

**Review by rileykarson — COMMENTED** [review]
Sorry, I lost this email (apparently multiple times) @ScottSuarez! Yeah, this is what I think will (ultimately) work.

However, I think this will have slightly unintended consequences based on https://github.com/hashicorp/terraform-provider-google-beta/blob/38006e725a1f086c78d8d252433c14fb1df9fb34/google-beta/resource_container_cluster.go#L3831-L3842. We'll probably need to check in that function whether autopilot is enabled and skip the check if so. It's possible we'd need to add a third case, and send a distinct method body.

The original bug report suggested the following config:

```
resource "google_container_cluster" "primary" {
  name             = "${var.project_id}-gke"
  location         = var.region
  enable_autopilot = true
}
```

We should add a test that verifies that an equally sparse configuration can be applied successfully. I think without changing this function that would still fail, and the workaround Cameron shared in https://github.com/hashicorp/terraform-provider-google/issues/10782#issuecomment-1024488630 would still be required.

(our autopilot tests are _really_ sparse, and the feature has evolved substantially on the API side since we added it. Preexisting green tests are not enough for it at our current level of coverage, unfortunately 🙁 )

**Comment by rileykarson** [comment]
@ScottSuarez, let me know when this is ready for review.

**Comment by ScottSuarez** [comment]
Will do, sorry for the noise, just trying to get the test in a good to go state. Couldn't get my local env to work with it for whatever reason.

**Comment by ScottSuarez** [comment]
@rileykarson ready for review

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -7661,3 +7682,12 @@ func TestValidateNodePoolAutoConfig(t *testing.T) {
 	}
 }
 <% end -%>
+
+func testAccContainerCluster_autopilot_minimal(name string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name             = "%s"
+  location         = "us-central1"
+	enable_autopilot = true
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:7691]:
```suggestion
  enable_autopilot = true
```

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 2]
```diff
@@ -7573,3 +7594,99 @@ resource "google_container_cluster" "primary" {
...
+func testAccContainerCluster_autopilot_minimal(name string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name             = "%s"
+  location         = "us-central1"
+  enable_autopilot = true
+}`, name)
+}
+=======
+>>>>>>> 957f0c2843b207f94432e7a868bc996372a3398a
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:7692]:
```suggestion
```

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 3]
```diff
@@ -7573,3 +7594,13 @@ resource "google_container_cluster" "primary" {
...
 }
 <% end -%>
+
+
+func testAccContainerCluster_autopilot_minimal(name string) string {
+	return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name             = "%s"
+  location         = "us-central1"
+	enable_autopilot = true
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:7604]:
```suggestion
  enable_autopilot = true
```

**Pull request merged**